datasets/me/acp_peps/crawler.py

```diff
@@ -0,0 +1,162 @@
+import io
+from csv import DictReader
+from datetime import datetime
+from typing import List, Dict, Optional, Tuple
+from zipfile import ZipFile, BadZipFile
+
+from zavod import Context, helpers as h
+from zavod.logic.pep import categorise
+from zavod.shed.trans import (
+    apply_translit_full_name,
+    make_position_translation_prompt,
+)
+
+TRANSLIT_OUTPUT = {
+    "eng": ("Latin", "English"),
+}
+POSITION_PROMPT = prompt = make_position_translation_prompt("cnr")
+
+
+def extract_latest_filing(
+    dates: List[Dict[str, object]]
+) -> Tuple[Optional[str], Optional[int]]:
+    if not dates:
+        return None, None
+
+    # Sort the dates in descending order
+    dates_sorted = sorted(
+        dates,
+        key=lambda item: datetime.strptime(item["datum"], "%Y-%m-%d"),
+        reverse=True,
+    )
+
+    # Assume the first entry is the latest
+    latest_entry = dates_sorted[0]
+    latest_date: Optional[str] = latest_entry.get("datum")
+    latest_report_id: Optional[int] = latest_entry.get("stariIzvjestaj")
+
+    # Check if the latest report is -1, move to the next valid entry
+    if latest_report_id == -1:
+        for entry in dates_sorted[1:]:
+            if entry.get("stariIzvjestaj") != -1:
+                latest_date = entry.get("datum")
+                latest_report_id = entry.get("stariIzvjestaj")
+                break
+        else:
+            # If all entries have -1, set the value to None
+            latest_report_id = None
+
+    return latest_date, latest_report_id
+
+
+def fetch_and_process_csv(context: Context, latest_report_id: int):
```

Comment:
I would just call this fetch_csv_rows - process sounds like we're doing a bit more with it than extracting it from the csv